### ISSUE TITLE
Allow to apply a patch from a package dependency

### ISSUE THREAD

**Issue by alfaproject** [pnpm/pnpm#7599]
### Contribution

- [ ] I'd be willing to implement this feature ([contributing guide](https://github.com/pnpm/pnpm/blob/main/CONTRIBUTING.md))

### Describe the user story

We have some patches that we share across multiple projects so we decided to move them to a shared 'infra' dependency which we then add as part of everyone of our projects.

We are now migrating from Yarn to pnpm and we used to use `patch-package` to apply our patches, and we could do this postinstall command:
```
"postinstall": "patch-package --error-on-warn && patch-package --error-on-warn --patch-dir=./node_modules/@the-mill/infra-serverless/patches"
```

That would then apply the local packages and the shared ones from the infra dependency but that isn't possible anymore with pnpm native patch mechanism due to the fact that it doesn't even let install dependencies before applying the patches )':

### Describe the solution you'd like

Basically allow to install project dependencies before applying patches and allow to patch local dependencies from a patch folder from a dependency (or some other mechanism that allows for that without having to fork projects)

### Describe the drawbacks of your solution

I'm not sure there are any without having to fork some projects and maintain said fork

### Describe alternatives you've considered

Forking

**Comment by zkochan** [comment]
You can do it now with [configurational dependencies](https://pnpm.io/config-dependencies#loading-patches).
